lessons/r/intro/lesson.Rmd

```diff
@@ -0,0 +1,555 @@
+
+```{r}
+airquality %>% 
+  arrange(Temp, Wind)
+```
+
+Finally, we can sort in descending order using the `desc()` helper function:
+
+```{r}
+airquality %>% 
+  arrange(desc(Month))
+```
+
+## `mutate` - apply windowed functions
+
+So far, we've explored functions that allow us to subset data frames and modify them
+as they are. Where `dplyr` arguably comes to life is its ability to actually perform
+vectorized operations across entire columns of a given data frame via the `mutate` function.
+
+The `Temp` column in our data frame contains temperature values in Fahrenheit. How might
+we convert these values to Celsius instead?
+
+Recall that an approximate conversion between the two is as follows:
+
+$$ C = (F - 32) / 1.8 $$
+Let's perform this conversion with `mutate`:
+
+```{r}
+airquality %>% 
+  mutate(temp_celsius = (Temp - 32) / 1.8)
+```
+
+For each value of `Temp` in each row, we've now created a new column called `temp_celsius` that
+stores those values converted to Celsius. Pretty neat, right?
+
+`mutate` can also be used to denote whether certain values meet a condition, in combination
+with a helper function called `ifelse`. Let's say we denote any observation with temp_celsius < 15
+as 'chilly'. We can encode that as a new variable in our data frame:
+
+```{r}
+airquality %>% 
+  mutate(temp_celsius = (Temp - 32) / 1.8) %>% 
+  mutate(chilly = ifelse(temp_celsius < 15, 1, 0))
+```
+
+## `group_by` and `summarise` - summary functions, grouped and ungrouped
+
+Finally, `dplyr` also allows us to compute simple summary statistics in a data frame
+using `summarise`. The syntax for `summarise` is similar to that of `mutate`, and also
+returns a data frame.
+
+```{r}
+airquality %>% 
+  summarise(mean_wind = mean(Wind), mean_temp = mean(Temp))
+```
+
+Where summarise becomes especially powerful, however, is when it's used alongside `group_by`
+to perform grouped operations. Let's say we wanted to calculate the above means (Wind and Temp)
+for each month individually. This is where `group_by` comes in:
+
+```{r}
+airquality %>% 
+  group_by(Month) %>% 
+  summarise(mean_wind = mean(Wind), mean_temp = mean(Temp))
+```
+
+`group_by` doesn't outwardly change anything about the data frame, but adds a bit of 'under the hood'
+information so that R knows any subsequent operations should be performed with those groupings in mind.
+
+# Reading and writing files
+
+Of course, our own data files are unlikely to be preloaded into R. For that reason, let's 
+briefly cover how to read and write files from and to our computers.
+
+These are made particularly easy using the `readr` package, which allow us
+to write our data frames to a csv file.
+
+```{r}
+library(readr)
+```
+
+First, let's check our current working directory. This is where R will both look for
+requested files and write new ones.
+
+
+```{r}
+getwd()
+# setwd() - change dir
+```
+
+Next, we can write `airquality` to file as follows:
```

Comment:
Here is a good spot for Rprojects. We can show that the result of `getwd()` will be different for everybody, then have them create an Rproject before saving the csv. `getwd()` will then return the root of the project and when the csv is saved it will be in this project's root (or maybe in a *data/* folder).

Reply:
This might disrupt the flow of the lesson since it 'resets' RStudio - let's instead have them make the project at the very beginning and just explain the wd point verbally?